Gems/Atom/Feature/Common/Code/Source/Stars/StarsFeatureProcessor.cpp

```diff
@@ -0,0 +1,201 @@
+
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RPI.Public/Shader/Shader.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/View.h>
+#include <Atom/RPI.Public/ViewportContext.h>
+#include <Atom/RPI.Public/ViewportContextBus.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        void StarsFeatureProcessor::Reflect(ReflectContext* context)
+        {
+            if (auto* serializeContext = azrtti_cast<SerializeContext*>(context))
+            {
+                serializeContext
+                    ->Class<StarsFeatureProcessor, FeatureProcessor>()
+                    ->Version(1);
+            }
+        }
+
+        void StarsFeatureProcessor::Activate()
+        {
+            m_sceneSrg = GetParentScene()->GetShaderResourceGroup();
+
+            const char* shaderFilePath = "Shaders/stars/stars.azshader";
+            m_shader = RPI::LoadCriticalShader(shaderFilePath);
+
+            m_drawListTag = m_shader->GetDrawListTag();
+
+            m_starsDataBufferIndex.Reset();
+
+            // do not create the pipeline state here, the scene has no pipelines at this point
+        }
+
+        void StarsFeatureProcessor::Deactivate()
+        {
+            m_sceneSrg = nullptr;
+            m_shader = nullptr;
+        }
+
+        void StarsFeatureProcessor::Simulate(const FeatureProcessor::SimulatePacket& packet)
+        {
+            AZ_PROFILE_SCOPE(RPI, "StarsFeatureProcessor: Simulate");
+            AZ_UNUSED(packet);
+
+            if (!m_enabled)
+            {
+                return;
+            }
+
+            const AZ::RPI::ViewportContextPtr viewportContext = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get()->GetDefaultViewportContext();
+            const AzFramework::WindowSize windowSize = viewportContext->GetViewportSize();
+            const float vpWidth = static_cast<float>(windowSize.m_width);
+            const float vpHeight = static_cast<float>(windowSize.m_height);
+            const float size = m_radiusFactor * AZStd::min<float>(1.f, AZStd::min<float>(vpWidth / 1280.f, vpHeight / 720.f));
+
+            AZ::ScriptTimePoint timePoint;
+            TickRequestBus::BroadcastResult(timePoint, &TickRequestBus::Events::GetTimeAtCurrentTick);
+
+            m_starsData[0] = size / vpWidth;
+            m_starsData[1] = size / vpHeight;
+            m_starsData[2] = m_intensityFactor * AZStd::min(1.f, size);
+            m_starsData[3] = static_cast<float>(timePoint.GetSeconds()) * 0.5f;
+
+            m_sceneSrg->SetConstant(m_starsDataBufferIndex, m_starsData);
+            m_sceneSrg->SetConstant(m_starsRotationMatrixIndex, m_orientation);
+        }
+
+        void StarsFeatureProcessor::Render(const FeatureProcessor::RenderPacket& packet)
+        {
+            AZ_PROFILE_FUNCTION(AzRender);
+
+            if (!m_enabled || !m_starsVertexBuffer)
+            {
+                return;
+            }
+
+            if (!m_meshPipelineState)
+            {
```

Comment:
It seems that you are missing the callbacks to handle change in the pipeline.  This can lead to a crash if happen since you are caching the pipeline state. 
Here are the methods you might want to support:
            // RPI::SceneNotificationBus overrides ...
            void OnRenderPipelineAdded(RPI::RenderPipelinePtr renderPipeline) override;
            void OnRenderPipelineRemoved(RPI::RenderPipeline* renderPipeline) override;
            void OnRenderPipelinePassesChanged(RPI::RenderPipeline* renderPipeline) override;